### PR TITLE
add doc for apps api GA change

### PR DESCRIPTION
**Pull request by jennaxue-okta**
## Description:
-   adding `pushStatus` attribute to /api/v1/apps API
```
    "credentials": {
        "scheme": "EDIT_USERNAME_AND_PASSWORD",
        "userNameTemplate": {
            "template": "user.login",
            "type": "CUSTOM",
            "pushStatus": "PUSH".  // the added attribute 
        },
        "revealPassword": true,
        "signing": {}
    },
```
- **Is this PR related to a Monolith release?** Yes, must be released along with GA of changes. Release: 2021.7.0

### Resolves:

* OKTA-396834

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 1]
```diff
@@ -6360,13 +6360,14 @@ Applications that are configured with `BASIC_AUTH`, `BROWSER_PLUGIN`, or `SECURE
...
+| template                                | mapping expression for username         | String                           | TRUE     | `${source.login}` |           | 1024       |            |
+| type                                    | type of mapping expression              | `NONE`,  `BUILT_IN`, or `CUSTOM` | FALSE    | BUILT_IN          |           |            |            |
+| userSuffix                              | suffix for built-in mapping expressions | String                           | TRUE     | NULL              |           |            |            |
+| pushStatus <ApiLifecycle access="ea" /> | push username on update                 | `PUSH`, `DONT_PUSH`              | TRUE     | NULL              |           |            |            |
+
+> **Note:** You must use the `CUSTOM` type when defining your own expression that is not built-in. The `pushStatus` <ApiLifecycle access="ea" /> parameter is visible when type is `CUSTOM` and `SUPPORT_PUSH_STATUS_FOR_APPS_API` feature is enabled.
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:6370]:
```suggestion
> **Note:** You must use the `CUSTOM` type when defining your own expression that is not built-in. The `pushStatus` <ApiLifecycle access="ea" /> parameter is visible when type is `CUSTOM` and `SUPPORT_PUSH_STATUS_FOR_APPS_API` feature is enabled.
```

We don't mention feature flags in our public doc. Is there a name for this feature that we can use instead?

**Review by jakubvul-okta — CHANGES_REQUESTED** [review]

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md (hunk 1 above)
Comment (jennaxue-okta) [reply packages/@okta/vuepress-site/docs/reference/api/apps/index.md:6370]: oh yeah, we might make this a self service FF, can we use that name instead?

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 2]
```diff
@@ -6570,8 +6570,9 @@ Specifies the template used to generate a user's username when the application i
 | template   | mapping expression for username         | String                           | TRUE     | `${source.login}` |           | 1024       |            |
 | type       | type of mapping expression              | `NONE`,  `BUILT_IN`, or `CUSTOM` | FALSE    | BUILT_IN          |           |            |            |
 | userSuffix | suffix for built-in mapping expressions | String                           | TRUE     | NULL              |           |            |            |
+| pushStatus | push username on update                 | `PUSH`, `DONT_PUSH`              | TRUE     | `DONT_PUSH` for `CUSTOM` type |           |            |            |
 
-> **Note:** You must use the `CUSTOM` type when defining your own expression that is not built-in.
+> **Note:** You must use the `CUSTOM` type when defining your own expression that is not built-in. `pushStatus` is only effective for `CUSTOM` type.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:6575]:
```suggestion
> **Note:** You must use the `CUSTOM` type when you define your own expression that is not built-in. The `pushStatus` property is effective only for the `CUSTOM` type.
```

**Review by susanharper-okta — COMMENTED** [review]
a couple edits

**Review by colekurashige-okta — APPROVED** [review]
Looks good from the App Platform side of things.

**Review by mohamedelsherif-okta — APPROVED** [review]
🚢

**Pull request merged**